lib/classification_lifecycle.rb

```diff
@@ -1,82 +1,106 @@
 class ClassificationLifecycle
   class ClassificationNotPersisted < StandardError; end
 
-  attr_reader :classification
-
-  def initialize(classification)
-    @classification = classification
-  end
-
-  def queue(action)
+  def self.queue(classification, action)
     unless classification.persisted?
       message = "Background process called before persisting the classification."
       raise ClassificationNotPersisted.new(message)
     end
     ClassificationWorker.perform_async(classification.id, action.to_s)
   end
 
-  def transact!(&block)
+  def self.perform(classification, action)
```

Comment:
Ah, actually, removing this would make it harder to test, since the spec needs to set up a stub on `.new` to return a spy so we can test that `#execute` is called on the instance.

Ideally the lifecycle would just be the worker itself, but I don't know if that's easy.